### PR TITLE
GitWorkflow Guide

### PR DESCRIPTION
**Pull request by csengstock**
Workflow and helper tools/scripts to realize a Git workflow and a CI/CD process for SAP Data Intelligence.

### CODE REVIEW

**Comment by KaiRoesner** [comment]
I would suggest to include a reference to the Git Workflow Guide in the top level readme file.

File: GitWorkflow/Readme.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  * The [System Management Command-Line Client (vctl)](https://help.sap.com/viewer/41b069490705457e9426b112a3f052bd/Cloud/en-US)
+* **Git Repository** (e.g., on GitHub or GitHub for Enterprises) to keep the [solution artifacts tracked in Git](GitWorkflow.md)
+* **Build Server** (e.g., [Jenkins](https://jenkins.io/)) to run the vctl commands of the CI/CD pipeline
+
+<img src="img/cicd_flow_di.png" alt="Development Workflow" width="900"/>
+
+The end-to-end development workflow can be summarized as follows:
+
+1. **Development**: A developer will create/update/delete graphs, operators, or dockerfiles using the SAP Data Intelligence Pipeline Modeler. All changes are visible in the user workspace of SAP Data Intelligence (`/vhome`).
+2. **Git Usage**: The developer is using the [VSCode application](vscode-app) to interact with the Git repository (i.e., `git add <graph-artifact>`, `git commit`).
```
Comment (KaiRoesner) [inline GitWorkflow/Readme.md:28]: If all development is done in the Pipeline Modeler, then developers could also use [Github Desktop](https://desktop.github.com/) to interact with the Git repository.

File: GitWorkflow/vscode-app/Readme.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
+# VSCode Application for SAP Data Intelligence
+
+The VSCode application is a [Visual Studio Code](https://code.visualstudio.com/) extension for
```
Comment (KaiRoesner) [inline GitWorkflow/vscode-app/Readme.md:3]: The distinction and (legal) relationship between this DI app and the standalone PC product should be made clearer. Especially it should be clarified who (SAP or MS) provides exactly what at which SLA. Or is this clarified elsewhere, if so where?

**Comment by csengstock** [comment]
Hi Kai,

thanks for the suggestions. Will update the top-level Readme and clarify the license of the VSC app.

File: GitWorkflow/vscode-app/Readme.md (hunk 2 above)
Comment (csengstock) [reply GitWorkflow/vscode-app/Readme.md]: Added disclaimer to the vscode readme

**Pull request merged**